File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/3082/BusinessApplication/Helpers/NotOperatorValueConverter.vb
Imports System
Imports System.Windows.Data

''' <summary>
''' <see cref="IValueConverter"/> bidireccional que permite enlazar el inverso de una propiedad boolean a una dependency property.
''' </summary>
Public Class NotOperatorValueConverter
    Implements IValueConverter


    ''' <summary>
    ''' Convierte el <paramref name="value"/> proporcionado en su inverso.
    ''' </summary>
    ''' <param name="value">Valor <c>Boolean</c> que se va a convertir.</param>
    ''' <param name="targetType">Es el type al que se va a convertir (omitido).</param>
    ''' <param name="parameter">Parámetro opcional (omitido).</param>
    ''' <param name="culture">Referencia cultural de la conversión (omitida).</param>
    ''' <returns>Inverso del <paramref name="value"/> de entrada.</returns>
    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Return Not DirectCast(value, Boolean)
    End Function

    ''' <summary>
    ''' Inverso de <see cref="Convert"/>.
    ''' </summary>
    ''' <param name="value">Valor que se va a revertir.</param>
    ''' <param name="targetType">Es el type al que se va a convertir (omitido).</param>
    ''' <param name="parameter">Parámetro opcional (omitido).</param>
    ''' <param name="culture">Referencia cultural de la conversión (omitida).</param>
    ''' <returns>Inverso del <paramref name="value"/> de entrada.</returns>
    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return Not DirectCast(value, Boolean)
    End Function
End Class

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/2052/BusinessApplication/Helpers/NotOperatorValueConverter.cs
namespace $safeprojectname$
{
    using System;
    using System.Windows.Data;

    /// <summary>
    /// 双向 <see cref="IValueConverter"/>，使您可以将 boolean 属性的反向值绑定到依赖属性。
    /// </summary>
    public class NotOperatorValueConverter : IValueConverter
    {
        /// <summary>
        /// 将给定 <paramref name="value"/> 转换为其反向值。
        /// </summary>
        /// <param name="value">要转换的<c>bool</c>值。</param>
        /// <param name="targetType">要转换到的 type (已忽略)。</param>
        /// <param name="parameter">可选参数(已忽略)。</param>
        /// <param name="culture">转换的区域性(已忽略)。</param>
        /// <returns>输入 <paramref name="value"/> 的反向值。</returns>
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return !((bool)value);
        }

        /// <summary>
        /// <see cref="Convert"/> 的反向值。
        /// </summary>
        /// <param name="value">要转换回的值。</param>
        /// <param name="targetType">要转换到的 type (已忽略)。</param>
        /// <param name="parameter">可选参数(已忽略)。</param>
        /// <param name="culture">转换的区域性(已忽略)。</param>
        /// <returns>输入 <paramref name="value"/> 的反向值。</returns>
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return !((bool)value);
        }
    }
}
